src/NUnitEngine/nunit.engine/Internal/SettingsStore.cs

```diff
@@ -96,38 +96,46 @@ public void LoadSettings()
 
         public void SaveSettings()
         {
-            if (_writeable)
+            if (_writeable && _settings.Keys.Count > 0)
```

Comment:
Is there ever a case where the settings file contains settings, but this run we want to write a new version with no settings? If not, then this looks good.